manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
```

Comment:
I coded this up, and it certainly simplifies the changes to this repo. Is it OK that for non-layers, the URLs field is just ignored by all the clients? It seems like overkill to add validation that len(URLs) == 0 for everything but layers...